tests/Integration/Encryption/EncryptionTest.php

```diff
@@ -2,34 +2,80 @@
 
 namespace Illuminate\Tests\Integration\Encryption;
 
+use Illuminate\Contracts\Encryption\Encrypter as EncrypterContract;
+use Illuminate\Encryption\Cipher;
+use Illuminate\Encryption\KeyStoreFactory\KeyRotationKeyStoreFactory;
 use RuntimeException;
 use Orchestra\Testbench\TestCase;
 use Illuminate\Encryption\Encrypter;
 use Illuminate\Encryption\EncryptionServiceProvider;
 
 class EncryptionTest extends TestCase
 {
-    protected function getEnvironmentSetUp($app)
-    {
-        $app['config']->set('app.key', 'base64:IUHRqAQ99pZ0A1MPjbuv1D6ff3jxv0GIvS2qIW4JNU4=');
-    }
-
-    protected function getPackageProviders($app)
-    {
-        return [EncryptionServiceProvider::class];
-    }
-
     public function test_encryption_provider_bind()
     {
+        $this->setUpEncryptionWithAppKey();
         self::assertInstanceOf(Encrypter::class, $this->app->make('encrypter'));
+        self::assertInstanceOf(Encrypter::class, $this->app->make(EncrypterContract::class));
     }
 
     public function test_encryption_will_not_be_instantiable_when_missing_app_key()
     {
+        $this->setUpEncryptionWithAppKey();
+
         $this->expectException(RuntimeException::class);
 
         $this->app['config']->set('app.key', null);
 
         $this->app->make('encrypter');
     }
+
+    public function test_encryption_will_work_with_the_KeyRotationKeyStoreFactory()
+    {
+        //Given
+        $keyIdUsedToEncrypt = $this->setUpRotationKeyEncryption();
+
+        //When
+        $this->app->register(EncryptionServiceProvider::class);
+        /** @var EncrypterContract $encrypter */
+        $encrypter = $this->app->make(EncrypterContract::class);
+        $encryptedValue = $encrypter->encrypt('test');
+        $encryptedPayload = json_decode(base64_decode($encryptedValue));
+
+        //Then
+        $this->assertEquals($keyIdUsedToEncrypt, $encryptedPayload->keyId);
```

Comment:
Shouldn't we assert against the encrypted payload somebody. The only assertion this test does is against the keyId.